src/Supertype.ts

```diff
@@ -0,0 +1,147 @@
+import {ObjectTemplate} from './ObjectTemplate';
+import * as serializer from './serializer';
+
+function constructorName(constructor) {
+    var namedFunction = constructor.toString().match(/function ([^(]*)/);
+    return namedFunction ? namedFunction[1] : null;
+}
+
+export type Constructable = new (...args: any[]) => {};
+
+
+/**
+ * This is the base class for typescript classes.  It must
```

Comment:
was there supposed to be more to this comment that got truncated accidentally?

Reply:
hahaha yep that's what it looks like.

Reply:
I'll just delete the it must part